File: Week 2/C Programming/prog5.c
#include <stdio.h>

int main() {
    int rows, cols;
    printf("Enter rows and columns: ");
    scanf("%d %d", &rows, &cols);

    int mat[rows][cols];
    printf("Enter matrix elements:\n");
    int zeroCount = 0;

    for (int i = 0; i < rows; i++) {
        for (int j = 0; j < cols; j++) {
            scanf("%d", &mat[i][j]);
            if (mat[i][j] == 0) {
                zeroCount++;
            }
        }
    }

    int totalElements = rows * cols;

    if (zeroCount > totalElements / 2) {
        printf("The matrix is sparse.\n");
    } else {
        printf("The matrix is not sparse.\n");
    }

    return 0;
}

File: Week 2/python Programming/prog5.py
def is_sparse(matrix):
    rows = len(matrix)
    cols = len(matrix[0]) if rows > 0 else 0

    zero_count = 0
    total_elements = rows * cols

    for row in matrix:
        zero_count += row.count(0)

    return zero_count > total_elements / 2

# Example:
matrix = [
    [0, 0, 3],
    [0, 0, 6],
    [7, 8, 0]
]

if is_sparse(matrix):
    print("The matrix is sparse.")
else:
    print("The matrix is not sparse.")
